src/JWT.php

```diff
@@ -199,10 +199,16 @@ public static function decode(
     public static function encode(
         array $payload,
         $key,
-        string $alg,
+        ?string $alg = null,
         string $keyId = null,
         array $head = null
     ): string {
+        if (is_a($key, Key::class)) {
```

Comment:
nit: this library prefers the use of `instanceof`
```suggestion
        if ($key, instanceof Key::class) {
```